In [2]:
import tensorflow as tf

from tensorflow.python.keras.models import Model
from tensorflow.python.keras.layers import Input,Reshape,Conv2D,MaxPooling2D,Flatten,Dense
from mnist import MNIST
data = MNIST(data_dir="data/MNIST/")


img_size = data.img_size
img_size_flat = data.img_size_flat
img_shape_full  = data.img_shape_full
fc_size = 128
num_classes = 10

inputs = Input(shape = (img_size_flat,))

net = inputs
net = Reshape(img_shape_full)(net)
net = Conv2D(kernel_size = 5 ,strides = 1, filters =16 ,padding  = 'same',activation = 'relu' )(net)
net = MaxPooling2D(pool_size = 2 ,strides = 2)(net)
net = Conv2D(kernel_size = 5 ,strides = 1, filters =36 ,padding  = 'same',activation = 'relu' )(net)
net = MaxPooling2D(pool_size = 2 ,strides = 2)(net)
net = Flatten()(net)
net = Dense(fc_size,activation = 'relu')(net)
outputs = Dense(num_classes,activation = 'softmax')(net)

from tensorflow.python.keras.optimizers import Adam
optimus = Adam(lr = 1e-3)


model_sam = Model(inputs = inputs ,outputs = outputs )
model_sam.compile(optimizer =optimus , loss = 'categorical_crossentropy' , metrics = (['accuracy']))
model_sam.fit(x = data.x_train , y = data.y_train , epochs = 3,batch_size = 64)

result = model_sam.evaluate(x = data.x_test , y = data.y_test)

Epoch 1/3
55000/55000 [==============================] - 35s 642us/sample - loss: 0.1739 - acc: 0.9477
Epoch 2/3
55000/55000 [==============================] - 36s 650us/sample - loss: 0.0499 - acc: 0.9849
Epoch 3/3
10000/10000 [==============================] - 2s 221us/sample - loss: 0.0293 - acc: 0.9899


# Accuracy

**It was found out to be a whooping 98.99%**